test/experimental/features/kreference_group/channel_to_channel.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kreference_group
+
+import (
+	"context"
+
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	"knative.dev/eventing/test/rekt/resources/channel"
+	"knative.dev/eventing/test/rekt/resources/subscription"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+)
+
+// ChannelToChannel tests a scenario where the flow is source -> A -> B -> sink
+// and A -> B subscription uses the KReference.Group field.
+func ChannelToChannel() *feature.Feature {
+	f := feature.NewFeature()
+
+	channelGVK := channel.GVK()
+	channelGVR := channel.GVR()
+	channelGroup := channelGVK.GroupKind().Group
+	channelAPIVersion, imcKind := channelGVK.ToAPIVersionAndKind()
+
+	channelAName := feature.MakeRandomK8sName("channel-a")
```

Comment:
This code is much more clear if you use the prober code. See the broker/channel conformance tests for an example.